### ISSUE TITLE
[ENHANCEMENT] Network access on jail startup

### ISSUE THREAD

**Issue by cpr-fbsd18** [BastilleBSD/bastille#338]
**Is your feature request related to a problem? Please describe.**
A jail with nginx takes a long time to start. With OCSP stapling enabled the startup blocks on name lookup.
Examining start.sh, the IP address of the jail is added to the <jails> table of PF only after startup causing the lookup to fail.

**Describe the solution you'd like**
Move the addition to the <jails> table to immediately before the jail call.

**Describe alternatives you've considered**
I worked around this by adding a static NAT rule for this jail.

**Additional context**
Used bastille-0.8.20210115 from latest pkg with private networking.

**Comment by cedwards** [comment]
I wonder if this is a race condition with nginx attempting the lookup too quickly.

The reason we do the steps in this order is the query to determine the jail IP wasn't supported offline. ie; the jail needs to be running to determine its IP, which we then add to the table.

I have a number of nginx jails and haven't seen this issue. Perhaps I'm not using OCSP stapling.

**Comment by cpr-fbsd18** [comment]
OK, I see. Then it might be better for me to look into starting nginx only after jail startup.

**Comment by adyxax** [comment]
I encountered this issue, not with nginx specifically but it is a race all right.

The problem arises when using the `<jails>` pf table to nat outgoing traffic like the documentation suggests, that is why @cpr-fbsd18 managed to work around the issue with a static pf rule. In `usr/local/share/bastille/start.sh` we can see that bastille will start the jail and only later populate the `<jails>` table with the jail's ip. I tested a patch to reverse the order and think it fixes the issue cleanly, I will submit a pull request for review.
